packages/grpc-native-core/index.d.ts

```diff
@@ -0,0 +1,1214 @@
+     * @param service The service descriptor
+     * @param implementation Map of method names to method implementation
+     * for the provided service.
+     */
+    addService(service: Service, implementation: { [name: string]: handleCall }): void;
+
+    /**
+     * Add a proto service to the server, with a corresponding implementation
+     * @deprecated Use `Server#addService` instead
+     * @param service The proto service descriptor
+     * @param implementation Map of method names to method implementation
+     * for the provided service.
+     */
+    addProtoService(service: Service, implementation: { [name: string]: handleCall }): void;
+
+    /**
+     * Binds the server to the given port, with SSL disabled if creds is an
+     * insecure credentials object
+     * @param port The port that the server should bind on, in the format
+     * "address:port"
+     * @param creds Server credential object to be used for SSL. Pass an
+     * insecure credentials object for an insecure port.
+     */
+    bind(port: string, creds: ServerCredentials): void;
+  }
+
+  /**
+   * An object that completely defines a service.
+   * @typedef {Object.<string, grpc~MethodDefinition>} grpc~ServiceDefinition
+   */
+  export interface ServiceDefinition {
+    [s: string]: MethodDefinition;
+  }
+
+  /**
+   * An object that completely defines a service method signature.
+   */
+  export interface MethodDefinition {
+    /**
+     * The method's URL path
+     */
+    path: string;
+    /**
+     * Indicates whether the method accepts a stream of requests
+     */
+    requestStream: boolean;
+    /**
+     *  Indicates whether the method returns a stream of responses
+     */
+   responseStream: boolean;
+   /**
+    * Serialization function for request values
+    */
+    requestSerialize: serialize;
+    /**
+     * Serialization function for response values
+     */
+    responseSerialize: serialize;
+    /**
+     * Deserialization function for request data
+     */
+    requestDeserialize: deserialize;
+    /**
+     * Deserialization function for repsonse data
+     */
+    responseDeserialize: deserialize;
+  }
+
+  type handleCall = handleUnaryCall | handleClientStreamingCall | handleServerStreamingCall | handleBidiStreamingCall;
+
+  /**
+   * User-provided method to handle unary requests on a server
+   */
+  type handleUnaryCall = (call: ServerUnaryCall, callback: sendUnaryData) => void;
+
+  export class ServerUnaryCall {
+    /**
+     * Indicates if the call has been cancelled
+     */
+    cancelled: boolean;
+
+    /**
+     * The request metadata from the client
+     */
+    metadata: Metadata;
+
+    /**
+     * The request message from the client
+     */
+    request: any;
+
+    /**
+     * An EventEmitter. Used for unary calls.
+     * @param call The call object associated with the request
+     * @param metadata The request metadata from the client
+     */
+    constructor(call: Call, metadata: Metadata);
```

Comment:
If this isn't necessary, I would prefer to omit it. This constructor is not explicitly exposed in the API. In addition, the `Call` type is defined elsewhere as the union of the client call object types, but that is not the correct type for this argument. The argument to this constructor is another type that is also not exposed in the API.

Reply:
I'll remove them.